### DISCUSSION TITLE
Adding new traffic_calming node/POI

### DISCUSSION THREAD

**Discussion by vizsim** [Helium314/SCEE#623] [Ideas]
Hi,
I was wondering if it was possible to add the ambility to add [traffic_calming](https://wiki.openstreetmap.org/wiki/Key:traffic_calming).
In my understanding it should be similar to [bollard](https://wiki.openstreetmap.org/wiki/Tag:barrier%3Dbollard) and bollard is possible in SCEE already.

**Comment by rusty-snake** [comment]
Yes, add a node to a way and search for 'speed bump' for example.

**Comment by Helium314** [comment]
You can add any node. If there is no suitable preset available, you can simply edit the tags.

**Comment by vizsim** [comment]
thanks guys!
sorry i am new to SCEE. I ended up adding a new node on top of the existing crossing node.
Can I also edit an exiting node instead of adding a new node?

**Reply by rusty-snake** [reply]
Yes, you can select nodes in a custom overlay (see "All nodes" in https://github.com/Helium314/SCEE/discussions/574 for showing virtually everything) and edit the tags of this node.
If there is already a `crossing` mapped that you want to edit you can also use the "Crossing" overlay.

**Reply by Helium314** [reply]
When you insert a node into a way on top of an existing node, the existing node should be selected automatically.

**Reply by rusty-snake** [reply]
@Helium314 this seems to not work correctly under certain unknown conditions. I used it IDK around 5 times to "select" an existing node (creating new nodes worked so far) and edit it. Two of those changesets got fixed up "self-intersection" (JSOM Validator). Since then I use custom overlays.

**Reply by mnalis** [reply]
@rusty-snake do you have an example of such node? It would be interesting to looks at the history and what actually happened there (i.e. was problem existing there also before SCEE, or did SCEE itself create the problem)

**Self-intersection** AFAIK typically refers to ***one*** single node, which gets used by same way twice (e.g. like [here](https://help.openstreetmap.org/questions/68631/how-to-remove-a-repeated-not-duplicate-node-present-in-a-same-way)). I.e. making the way cross over itself (passing through the same node).

It there was another node being added with same coordinates, it would typically be called **duplicate node** (like e.g. on [OSMI](https://tools.geofabrik.de/osmi)).

While it is possible that SCEE has a bug which causes the same node id to be used in a way twice (which is why example would be very useful, if you can provide it!), it does not mean that new node id is being added to the same position.

**Reply by rusty-snake** [reply]
> passing through the same node

Maybe something like this?:

```
Node1 -->-->-- Node2
                | |
                v ^
                | |
               Node3
```

At least that's the only way I can think of a "self intersection" (I edited Node2).

> It would be interesting to looks at the history and what actually happened there (i.e. was problem existing there also before SCEE, or did SCEE itself create the problem)

Statistically SCEE. I edit a lot with SC(EE) and iD but only this SCEE feature has "caused" such follow-up changesets. In 20-30% of the times I used it.

> do you have an example of such node?

Sending on OSM as I don't want my OSM account directly linked to my github account.

**Reply by Helium314** [reply]
When you have some ideas what could cause the issue, or just want to check before uploading:
When you undo, you see the changes you did. If there is a text "you created a node with..." then really a new node is created instead of editing and existing node.

**Reply by vizsim** [reply]
wow, amazing, the custom "all nodes" overlay is all i need for now.
thanks for the help and anyway thanks for creating and maintaining this great app!!
